### ISSUE TITLE
Webpack error on Android with plugin added

### ISSUE THREAD

**Issue by hypery2k** [NativeScript/NativeScript#2760]
I'm facing issues when adding the [drawing-pad plugin](https://github.com/bradmartin/nativescript-drawingpad/issues/9) in combination with the webpack build. I'm always getting the following error:

```
09-19 12:44:06.025 19533 19533 E AndroidRuntime: TypeError: Cannot read property 'gcacace' of undefined
09-19 12:44:06.025 19533 19533 E AndroidRuntime: File: "/data/data/de.sample.app/files/app/bundle.js, line: 95825, column: 30
```

Any ideas?

I check a similar [issue](https://github.com/EddyVerbruggen/nativescript-plugin-firebase/issues/41), but it didn't helped

**Comment by hypery2k** [comment]
it seems to be related to ABIs split

**Comment by hypery2k** [comment]
If I add manually the dependency in my app.gradle it works:
```
dependencies {
  compile "com.github.gcacace:signature-pad:1.0.3"
}
```
Seems to be somekind of merge error with dependencies in app.gradle

**Comment by NickIliev** [comment]
ping @slavchev , @Pip3r4o

**Comment by Pip3r4o** [comment]
The NS plugin in question wraps a native (Android) plugin, and leverages NS's power to access native API. We expose metadata for all native API that is referenced in the project (some of which is used in the drawing pad plugin), and for whatever reason, we do not build Metadata, or the native plugin isn't downloaded in the first place. I will check it out

**Comment by jlooper** [comment]
Watching this one, and ping @nathanwalker

**Comment by Pip3r4o** [comment]
@hypery2k could you try removing the platforms/android dir and building the app again? The plugin should have a its dependency declared already in `configurations/nativescript-drawingpad/include.gradle` and thus should not require that any additional work be done to make it function

**Comment by hypery2k** [comment]
I always doing this ;) But it doesn't help

**Comment by hypery2k** [comment]
the issue seems to be coming from using es6 in the tsconfig:
```
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es6",
    "sourceMap": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "noEmitHelpers": true,
    "noEmitOnError": true
  },
  "compileOnSave": true,
  "filesGlob": [
    "./app/**/*.ts",
    "./references.d.ts"
  ],
  "exclude": [
    "node_modules",
    "platforms"
  ]
}

```
If I'm using es5 it's working. But seems a little bit strange to me. I can also use es6 if I adopt my references.d:
```
/// <reference path="./node_modules/tns-core-modules/tns-core-modules.es6.d.ts" /> Needed for autocompletion and compilation.
/// <reference path="./nativescript-permissions"/>
/// <reference path="./node_modules/nativescript-appinfo/appinfo"/>
/// <reference path="./node_modules/nativescript-drawingpad/index"/>
/// <reference path="./node_modules/@nextrust/angular-common/nextrust.d.ts"/>

interface Thenable<R> {
  then<U>(onFulfilled?: (value: R) => Thenable<U>, onRejected?: (error: any) => Thenable<U>): Thenable<U>;
  then<U>(onFulfilled?: (value: R) => Thenable<U>, onRejected?: (error: any) => U): Thenable<U>;
  then<U>(onFulfilled?: (value: R) => Thenable<U>, onRejected?: (error: any) => void): Thenable<U>;
  then<U>(onFulfilled?: (value: R) => U, onRejected?: (error: any) => Thenable<U>): Thenable<U>;
  then<U>(onFulfilled?: (value: R) => U, onRejected?: (error: any) => U): Thenable<U>;
  then<U>(onFulfilled?: (value: R) => U, onRejected?: (error: any) => void): Thenable<U>;
}
```
Can't I use ES6 normally with NativeScript? Not sure if this related to library error

**Comment by Pip3r4o** [comment]
@hypery2k The android runtime will bring ES6 support in the upcoming (2.4) release with the update of the javascript (v8) engine. I don't however understand  yet how that is causing the problems in question.

**Comment by hypery2k** [comment]
good to know. For us it's currently working with the gradle and tsconfig hacks.
Did you have a timeline for 2.4?

**Comment by NickIliev** [comment]
@hypery2k the issue is resolved with NativeScript 3.4.0 and higher version.
I have tested with next (4.0.0) with[ this demo application ](https://github.com/NickIliev/NS-Issues-2018/tree/master/NativeScript/issue_2760)which is using the mentioned plugin and bundling with WebPack is working as expected.

Notice that I still had to add the `nativescript-drawingpad` to `vendor.ts`
```
global.registerModule("nativescript-drawingpad", ()=> require('../node_modules/nativescript-drawingpad'))

```
